freezegun/api.py

```diff
@@ -322,7 +322,11 @@ def __sub__(self, other):
 
     @classmethod
     def today(cls):
-        result = cls._date_to_freeze() + cls._tz_offset()
+        if _should_use_real_time():
+            result = real_date.today() + cls._tz_offset()
+        else:
+            result = cls._date_to_freeze() + cls._tz_offset()
```

Comment:
I don't understand what this added logic is for. `cls._date_to_freeze()` calls `get_current_time()` which already calls `_should_use_real_time()` and does this type of thing so this change seems redundant?

https://github.com/spulec/freezegun/blob/92624e42130684163040b30a95a2c2e9e1cd7a4d/freezegun/api.py#L174

Reply:
@boxed I'm happy to submit my change which [should be smaller](https://github.com/spulec/freezegun/compare/master...immerrr:consult-ignore-list-in-datetime-now-date-today?expand=1) if that would make it easier to get this in.

I could add the test case from this PR to make sure it covers the issue mentioned in this PR, too.

Reply:
Just to make sure we are talking about the same thing, you mean my patch that touches these two conditionals is too complex, right?

![image](https://user-images.githubusercontent.com/579798/157917768-0e45fe3d-25b7-41dc-b352-b234d7006c5f.png)

Reply:
Yes. Correct. I feel like this logic should be fully contained within get_current_time itself. It just wouldn't be DRY otherwise, and it would also mean there's a ton of duplicate code and every place there isn't duplication would be a bug.

Reply:
There are also performance considerations here, e.g. some blocks like 
```
if _should_use_real_time():
    return call_some_builtin_function()
```
may work much faster than converting the result of `get_current_time()` back to the required format.

DRY is a valid principle, but repetition only manifests itself if the code needs to change. If it is a small module that is feature complete, what difference does it make if the code repeats one time, two times or ten times, the interpreter doesn't care.

That said, let me see if I can hide `should_use_real_time` inside `get_current_time`...

Reply:
> may work much faster than converting the result of get_current_time() back to the required format.

There is no conversion in this code though.

Reply:
> There is no conversion in this code though.

There are conversions in functions like `fake_monotonic` whose "faking" versions use `get_current_time`

https://github.com/spulec/freezegun/blob/be4127bff37682591b0b099b9a2904689133969b/freezegun/api.py#L221-L225

> but get_current_time already calls should_use_real_time! 

I don't think it does, or maybe i'm looking somewhere else

https://github.com/spulec/freezegun/blob/be4127bff37682591b0b099b9a2904689133969b/freezegun/api.py#L169-L170